src/test/java/com/google/sps/UserLoginTest.java

```diff
@@ -0,0 +1,95 @@
+package com.google.sps;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.sps.servlets.UserLoginServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class UserLoginTest {
+
+  @Mock private HttpServletRequest request;
+
+  @Mock private HttpServletResponse response;
+
+  private UserLoginServlet userServlet =
+    new UserLoginServlet();  
+
+  private LocalServiceTestHelper localHelper =
+    new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+        .setEnvIsAdmin(true);
+  
+  
+  private JsonObject getLoginServletResponse() throws ServletException, IOException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    /*
+    Mocking the print writer bc
+    UserLoginServlet writes its reponse in response.getWriter() 
+    */
+    when(response.getWriter()).thenReturn(printWriter);
+
+    // UserLoginServlet userServlet = new UserLoginServlet();
```

Comment:
It's best not to have commented out code checked in. maybe remove this?

Reply:
done